sdk/spring/spring-cloud-azure-feature-management/src/test/java/com/azure/spring/cloud/feature/management/validationstests/ValidationsTest.java

```diff
@@ -0,0 +1,140 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+package com.azure.spring.cloud.feature.management.validationstests;
+
+import com.azure.spring.cloud.feature.management.FeatureManager;
+import com.azure.spring.cloud.feature.management.filters.TargetingFilter;
+import com.azure.spring.cloud.feature.management.filters.TargetingFilterTestContextAccessor;
+import com.azure.spring.cloud.feature.management.filters.TimeWindowFilter;
+import com.azure.spring.cloud.feature.management.implementation.FeatureManagementConfigProperties;
+import com.azure.spring.cloud.feature.management.implementation.FeatureManagementProperties;
+import com.azure.spring.cloud.feature.management.validationstests.models.ValidationTestCase;
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.MapperFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.fasterxml.jackson.databind.type.CollectionType;
+import com.fasterxml.jackson.databind.type.TypeFactory;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.MockitoAnnotations;
+import org.springframework.context.ApplicationContext;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.util.LinkedHashMap;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+public class ValidationsTest {
+    @Mock
+    private ApplicationContext context;
+    @Mock
+    private FeatureManagementConfigProperties configProperties;
+
+    private final ObjectMapper OBJECT_MAPPER = JsonMapper.builder()
+        .configure(MapperFeature.ACCEPT_CASE_INSENSITIVE_PROPERTIES, true).build();
+    private final String TEST_CAE_FOLDER_PATH = "validations-tests";
+    private final String INPUTS_USER = "user";
+    private final String INPUTS_GROUPS = "groups";
+    private final String SAMPLE_FILE_NAME_FILTER = "sample";
+    private final String TESTS_FILE_NAME_FILTER = "tests";
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+        when(configProperties.isFailFast()).thenReturn(true);
+        when(context.getBean(Mockito.contains("TimeWindow"))).thenReturn(new TimeWindowFilter());
```

Comment:
We could make this more like an integration test and use `@SpringBootTest` I think that would make it so we don't have to Mock and make it so FeatureManager doesn't have to become public.

Reply:
If we use `@SpringBootTest`, then we need to define multi file? For exmaple, one test case file for "NoFilters", one test case file for "RequirementType"... 

```
@ExtendWith(SpringExtension.class)
@EnableConfigurationProperties(value = FeatureManagementProperties.class)
@TestPropertySource(value = { "classpath:/validations-tests/NoFilters.sample.json" }, factory= JsonPropertySourceFactory.class)
public class NoFiltersTest {

    @Autowired
    private FeatureManagementProperties properties;

}
```

```
@ExtendWith(SpringExtension.class)
@EnableConfigurationProperties(value = FeatureManagementProperties.class)
@TestPropertySource(value = { "classpath:/validations-tests/RequirementType.sample.json" }, factory= JsonPropertySourceFactory.class)
public class NoFiltersTest {

    @Autowired
    private FeatureManagementProperties properties;

}
```

Reply:
Yes, I think we have to. We could make a sub class they all extend of they have enough overlap,

Reply:
If we just want to avoid make feature manager constructor as a public method, we could move the `ValidationsTest` to another package, from `com.azure.spring.cloud.feature.management.validationstests` to `com.azure.spring.cloud.feature.management`.

I don't think it's a good idea to have multi sub class for test jsons. I think we want to download the test json files from the repo in the future, and if there're new test cases, **that would be better the test cases succeed without adding new files to feature management lib**. 
If we use `@TestPropertySource`, then we need to hard-code the configuration file name. So if we added a new file like "Variant.Sample.json", then we need to add a new sub test case class.